osutil/disks/verity.go

```diff
@@ -0,0 +1,50 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package disks
+
+import (
+	"path/filepath"
+	"strings"
+)
+
+func cryptVerityDeviceMapperBackResolver(opts *deviceMapperBackResolversOpts) (dev string, ok bool) {
+	if !strings.HasPrefix(string(opts.dmUUID), "CRYPT-VERITY") {
```

Comment:
This might crash if called with opts being nil due to a refactor that isn't obvious.

Let's add a check that opts is not nil, or pass opts by value (with a string and two slices it's probably okay).

Reply:
Added.